homeassistant/components/backup/manager.py

```diff
@@ -251,25 +246,27 @@ async def async_upload_backup(self, *, slug: str, **kwargs: Any) -> None:
         if not (backup := await self.async_get_backup(slug=slug)):
             return
 
+        local_agent = cast(LocalBackupAgent, self.backup_agents[LOCAL_AGENT_ID])
```

Comment:
I don't think we should treat local agents differently or be aware of specific agent ids.